support/refc/prim.c

```diff
@@ -82,7 +82,7 @@ Value *idris2_crash(Value *msg) {
 
 Value *idris2_Data_IOArray_Prims_prim__newArray(Value *erased, Value *_length,
                                                 Value *v, Value *_word) {
-  int length = extractInt(_length);
+  int length = idris2_vp_to_Int64(_length); // it sould Int32
```

Comment:
I searched for Value_Int and replaced it to idris2_vp_to_.

That comment is just a doodling. I meant that Int32 is better in terms of performance if it is wide enough, because Int64 isn't unboxable in 64bit CPU. And in many cases 32 bit width is sufficient.